src/ompl/geometric/planners/bitstar/datastructures/src/IntegratedQueue.cpp

```diff
@@ -0,0 +1,1679 @@
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+/* Authors: Jonathan Gammell */
+
+//Myself:
+#include "ompl/geometric/planners/bitstar/datastructures/IntegratedQueue.h"
+
+//OMPL:
+//For exceptions:
+#include "ompl/util/Exception.h"
+
+namespace ompl
+{
+    namespace geometric
+    {
+        BITstar::IntegratedQueue::IntegratedQueue(const VertexPtr& startVertex, const VertexPtr& goalVertex, const NeighbourhoodFunc& nearSamplesFunc, const NeighbourhoodFunc& nearVerticesFunc, const VertexHeuristicFunc& lowerBoundHeuristicVertex, const VertexHeuristicFunc& currentHeuristicVertex, const EdgeHeuristicFunc& lowerBoundHeuristicEdge, const EdgeHeuristicFunc& currentHeuristicEdge, const EdgeHeuristicFunc& currentHeuristicEdgeTarget)
+            :   opt_(startVertex->getOpt()),
+                startVertex_(startVertex),
+                goalVertex_(goalVertex),
+                nearSamplesFunc_(nearSamplesFunc),
+                nearVerticesFunc_(nearVerticesFunc),
+                lowerBoundHeuristicVertexFunc_(lowerBoundHeuristicVertex),
+                currentHeuristicVertexFunc_(currentHeuristicVertex),
+                lowerBoundHeuristicEdgeFunc_(lowerBoundHeuristicEdge),
+                currentHeuristicEdgeFunc_(currentHeuristicEdge),
+                currentHeuristicEdgeTargetFunc_(currentHeuristicEdgeTarget),
+                useFailureTracking_(false),
+                delayRewiring_(true),
+                outgoingLookupTables_(true),
+                incomingLookupTables_(true),
+                vertexQueue_( boost::bind(&BITstar::IntegratedQueue::vertexQueueComparison, this, _1, _2) ), //This tells the vertexQueue_ to use the vertexQueueComparison for sorting
+                vertexToExpand_( vertexQueue_.begin() ),
+                edgeQueue_( boost::bind(&BITstar::IntegratedQueue::edgeQueueComparison, this, _1, _2) ), //This tells the edgeQueue_ to use the edgeQueueComparison for sorting
+                vertexIterLookup_(),
+                outgoingEdges_(),
+                incomingEdges_(),
+                resortVertices_(),
+                costThreshold_( std::numeric_limits<double>::infinity() ), //Purposeful gibberish
+                hasSolution_(false)
+        {
+            //The cost threshold:
```

Comment:
One nitpick: In general, if the comment does not add much to what I would get from the variable name, I prefer to not add it.